stripe-core/src/main/java/com/stripe/android/core/networking/ConnectionFactory.kt

```diff
@@ -26,7 +27,10 @@ interface ConnectionFactory {
     @Throws(IOException::class, InvalidRequestException::class)
     fun createForFile(request: StripeRequest, outputFile: File): StripeConnection<File>
 
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
     object Default : ConnectionFactory {
+        val testConnectionCustomization = AtomicReference<((HttpURLConnection) -> Unit)?>(null)
```

Comment:
Why does this have to be an AtomicReference?

Reply:
We don't need AtomicReference here, but it's typically my default when I know I want things to be safe across threads.  (I was setting this from a test thread, and it was being read by whatever thread the network requests were happening on). Since we don't need more features of AtomicReference (like `compareAnd*`), I updated them to be volatile fields, which gives it the reads we need after setting from another thread, but doesn't have as much overhead.